cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,104 @@
-# Introduction
+# Forgot Password and MFA Cheat Sheet
 
-This article provides a simple model to follow when implementing a *forgot password* web application feature.
+## Introduction
 
-# The Problem
+In order to implement a proper user management system, almost all systems integrate a **Forgot Password/MFA?** service, which allows the user to request a reset to their password or MFA whenever they forget any of them, or if their account ever gets breached.
 
-There is no industry standard for implementing a **Forgot Password** feature. The result is that you see applications forcing users to jump through myriad hoops involving emails, special URLs, temporary passwords, personal security questions, and so on. In the end you have to reset it to a new value.
+As much as is this functionality looks straight-forward and easy to implement, the details of its implementation makes it a sweet spot for security attacks, such as the renowned [user enumeration attack](https://owasp.org/www-project-web-security-testing-guide/stable/4-Web_Application_Security_Testing/03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.html).
 
-# Steps
+The following short guidelines can be used as a quick reference to protect the forgot password or MFA service:
 
-## Step 1) Gather Identity Data or Security Questions
+- **Unified user message, whether the user exists or not.**
+- **Ensure that the time taken for the user response message is uniform.**
+- **Use a side-channel to communicate the method to reset their password or MFA.**
+- **For critical services and applications, MFA should be used.**
 
-The first page of a secure Forgot Password feature asks the user for multiple pieces of hard data that should have been previously collected (generally when the user first registers). 
+## Content
 
-Steps for this are detailed in the identity section of the [Choosing and Using Security Questions Cheat Sheet](Choosing_and_Using_Security_Questions_Cheat_Sheet.md#choosing-security-questions).
+- [Forgot Password and MFA Cheat Sheet](#forgot-password-and-mfa-cheat-sheet)
+  * [Introduction](#introduction)
+  * [Content](#content)
+  * [Methods](#methods)
+  * [Methods Implementation](#methods-implementation)
+    + [One Time Password](#one-time-password)
+    + [URL Tokens](#url-tokens)
+    + [Security Questions](#security-questions)
+    + [Backup Codes](#backup-codes)
+  * [Operational Tasks](#operational-tasks)
 
-At a minimum, you should have collected some data that will allow you to send the password reset information to some out-of-band side-channel, such as a (possibly different) email address or an SMS text number, etc. to be used in Step 3.
+## Methods
 
-## Step 2) Verify Security Questions
+In order to allow a user to request a password or MFA reset, you will need to have some way to identify the user, or a mean to reach out to them through a side-channel.
```

Comment:
"mean" here should be "means". As in the phrase "[a means to an end](https://www.merriam-webster.com/dictionary/a%20means%20to%20an%20end)".